### PR TITLE
Feat event organizer field

### PR DESCRIPTION
**Pull request by jonasdeluna**
# Description

Adds event organizer field to specify responsible users.

image

image

# Result

If you've made visual changes, please include **before** and **after** images, preferably with a description. Make sure they do not contain any real user information.

# Testing

- [x] I have thoroughly tested my changes.

Please describe what and how the changes have been tested, and provide instructions to reproduce if necessary.

---

Depends on: https://github.com/webkom/lego/pull/3496
Resolves: ABA-570

### CODE REVIEW

File: app/routes/events/EventCreateRoute.ts [hunk 1]
```diff
@@ -68,6 +70,8 @@ const mapStateToProps = (state, props) => {
     },
     pools: valueSelector(state, 'pools'),
   };
+  const currentUser: DetailedUser = selectCurrentUser(state);
```
Comment (ivarnakken) [inline app/routes/events/EventCreateRoute.ts:73]: `EventCreateRoute` (and therefore also `EventEditor`) has already access to `currentUser` through its props. See app/routes/events/index.tsx

Also, the correct type is `CurrentUser`, not `DetailedUser`, but that doesn't matter since you can just remove this.

File: app/routes/events/components/EventEditor/index.tsx [hunk 2]
```diff
@@ -306,6 +306,15 @@ function EventEditor({
               component={SelectInput.AutocompleteField}
               placeholder="Ansvar for arrangement"
             />
+            <Field
+              name="responsibleUsers"
+              label="Ansvarlige Brukere"
```
Comment (ivarnakken) [inline app/routes/events/components/EventEditor/index.tsx:311]:
```suggestion
              label="Ansvarlige brukere"
```

File: app/routes/events/utils.ts [hunk 3]
```diff
@@ -85,6 +85,7 @@ export type EditingEvent = Event & {
   hasFeedbackQuestion: boolean;
   isClarified: boolean;
   authors: Option[];
+  responsibleUsers: Option[];
```
Comment (ivarnakken) [inline app/routes/events/utils.ts:88]: And here it's a completely new type

File: app/store/models/Reaction.d.ts [hunk 4]
```diff
@@ -13,4 +13,5 @@ export interface ReactionsGrouped {
   count: number;
   hasReacted: boolean;
   reactionId?: ID;
+  user?: string;
```
Comment (ivarnakken) [inline app/store/models/Reaction.d.ts:16]: Is this related to this PR? We haven't merged in the backend changes for this yet

File: app/routes/articles/ArticleEditRoute.ts [hunk 5]
```diff
@@ -23,7 +24,7 @@ const mapStateToProps = (state, props) => {
   });
 
   const currentUser = selectCurrentUser(state);
-  const authors = article?.authors?.length
+  const authors: Array<DetailedUser> = article?.authors?.length
```
Comment (ivarnakken) [inline app/routes/articles/ArticleEditRoute.ts:27]: Please be consistent on this
```suggestion
  const authors: DetailedUser[] = article?.authors?.length
```
It's not _super_ important since it's equivalent, but still nice to be consistent

**Review by ivarnakken — COMMENTED** [review]
This looks mostly good, but I do have a question:

As I see it now, if the user has not selected any responsible users - it will default to the creator. But can't we make this optional instead? There are usually multiple "responsible people" for an event (sometimes many), and I'd hate for the detailed event page to get even more crowded with names. 

I interpreted the issue to be that authors would receive all questions related to events, instead of the responsible group. Therefore, wouldn't it be clean to make this optional? In other words; to only show responsible users if they want to be shown, or else show nothing (meaning only the responsible group). I think that would make the detailed event page a tiny bit more organized.

File: app/routes/events/components/EventDetail/index.tsx [hunk 6]
```diff
@@ -333,7 +334,7 @@ export default class EventDetail extends Component<Props, State> {
               <Link to={groupLink}>{event.responsibleGroup.name}</Link>
             ) : (
               event.responsibleGroup.name
-            )}{' '}
+            )}
```
Comment (norbye) [inline app/routes/events/components/EventDetail/index.tsx:337]: I'm on mobile so might've missed it, but have you added styling to replace this space between the group name and the email?

**Review by norbye — COMMENTED** [review]
Agree with @ivarnakken that it would be nice to **not** show the author if the field is not populated. IMO that would make more people send emails to the entire group when a persons name is not visible.

I'm not sure how many people are normally responsible for each event - but an alternative could also be to call the field "Kontaktperson" and make it singular (or "Kontaktpersoner" and plural). In that way it wouldn't feel as natural for the event creator to add all the responsible people to the list and add more clutter to the sidebar.

**Comment by jonasdeluna** [comment]
I've made it so that it's now optional and will display the author if the responsible users are not set. The responsible group will be shown at all times. Is this the behavior we want?

File: app/routes/events/EventCreateRoute.ts [hunk 7]
```diff
@@ -114,6 +116,7 @@ const mapStateToProps = (state, props) => {
       }),
       registrationDeadlineHours: 2,
       unregistrationDeadlineHours: 2,
+      responsibleUsers: [],
```
Comment (norbye) [inline app/routes/events/EventCreateRoute.ts:119]: To avoid making the process of creating new events more tiresome, we could set the default value of responsibleUsers to be the author here. In that way, if the responsible user is creating the event they wouldn't have to change anything, but they have the option to do so if they desire.

File: app/routes/events/components/EventDetail/index.tsx [hunk 8]
```diff
@@ -342,20 +343,46 @@ export default class EventDetail extends Component<Props, State> {
...
+                </Link>
+              ),
+            },
+          ]
+        : [
+            {
+              key: 'Forfatter',
+              value: 'Anonym',
+            },
+          ]),
```
Comment (norbye) [inline app/routes/events/components/EventDetail/index.tsx:384]: So our thought was to basically just remove this side of the ternary and keep the previous lines

**Review by norbye — APPROVED** [review]
> I've made it so that it's now optional and will display the author if the responsible users are not set. The responsible group will be shown at all times. Is this the behavior we want?

I think that what @ivarnakken and I meant would be the following;  
Always display the responsible group
Display responsible users if they are set, if not display nothing (alas, never display the author)

Anyways, imo this is pretty good, so I'd say it's good to go if you want it to be, then we can always iterate.

File: app/routes/events/components/EventDetail/index.tsx (hunk 8 above)
Comment (jonasdeluna) [reply app/routes/events/components/EventDetail/index.tsx:384]: Ok great I'll implement the changes and get it merged 👍

File: app/routes/events/EventCreateRoute.ts (hunk 7 above)
Comment (jonasdeluna) [reply app/routes/events/EventCreateRoute.ts:119]: This was how it was before the latest commit so I can revert it back to that 👍
